pkg/controllers/resources/persistentvolumes/translate.go

```diff
@@ -124,6 +129,10 @@ func (s *persistentVolumeSyncer) translateUpdate(ctx context.Context, vPv *corev
 	}
 
 	translatedStorageClassName := translateStorageClass(vPv.Spec.StorageClassName)
+	if !s.useFakePersistentVolumes {
+		translatedStorageClassName = vPv.Spec.StorageClassName
```

Comment:
Why is this necessary?

Reply:
because if we use real persistent volumes, they need to reference the proper storage class, for some reasons the storage class was updated to something different than the one at creation time when the persistent volume was updated